tools/restatectl/src/commands/log/reconfigure.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::num::{NonZeroU32, NonZeroU8};
+
+use anyhow::Context;
+use cling::prelude::*;
+use tonic::codec::CompressionEncoding;
+use tonic::transport::Channel;
+
+use restate_admin::cluster_controller::protobuf::cluster_ctrl_svc_client::ClusterCtrlSvcClient;
+use restate_admin::cluster_controller::protobuf::{
+    ListLogsRequest, ProviderKind as ProtoProviderKind, SealAndExtendChainRequest,
+};
+use restate_types::logs::metadata::{Logs, ProviderKind};
+use restate_types::logs::LogId;
+use restate_types::protobuf::common::Version;
+use restate_types::replicated_loglet::{NodeSet, ReplicatedLogletParams, ReplicationProperty};
+use restate_types::storage::StorageCodec;
+use restate_types::{GenerationalNodeId, PlainNodeId};
+
+use crate::app::ConnectionInfo;
+use crate::util::grpc_connect;
+
+#[derive(Run, Parser, Collect, Clone, Debug)]
+#[cling(run = "reconfigure")]
+pub struct ReconfigureOpts {
+    /// LogId/Partition to seal and extend
+    #[clap(long)]
+    log_id: u32,
+    /// Option segment index to seal, chosen automatically if not provided.
+    #[clap(long)]
+    segment_index: Option<u32>,
+    /// Min metadata version
+    #[clap(long, default_value = "1")]
+    min_version: NonZeroU32,
+    /// Provider kind.
+    #[clap(long, default_value = "replicated")]
+    provider: ProviderKind,
+
+    /// Replication property
+    /// If not provided, it will be copied from the last segment if (and only if) it was
+    /// of replicated type.
+    #[clap(long)]
+    replication_factor: Option<NonZeroU8>,
```

Comment:
This works for now, but in the future we'll need to update this to support location-aware format. Perhaps call this "replication" so we can evolve the type without changing the name?

Reply:
Changing from Option<NonZeroU8> to Option<String> is not a breaking change for the cmdline interface. So while I totally understand this is probably needed in the future to support more complex location aware replication property, I would suggest we leave it for a follow up PR.

Reply:
+1 to @pcholakov's approach. Let's go with this but it's okay to add `replication` in a later change, this should not block us.

Reply:
i will do the rename now and add `replication` later